patches/server/0949-Paper-Plugins.patch

```diff
@@ -0,0 +1,6590 @@
++        }
++
++        try {
++            return new PaperPluginClassLoader(logger, source, jarFile, configuration, this.getClass().getClassLoader(), new URLClassLoader(urls));
++        } catch (IOException exception) {
++            throw new RuntimeException(exception);
++        }
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/plugin/loader/library/PaperLibraryStore.java b/src/main/java/io/papermc/paper/plugin/loader/library/PaperLibraryStore.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..5fcce65009f715d46dd3013f1f92ec8393d66e15
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/plugin/loader/library/PaperLibraryStore.java
+@@ -0,0 +1,21 @@
++package io.papermc.paper.plugin.loader.library;
++
++import org.jetbrains.annotations.NotNull;
++
++import java.nio.file.Path;
++import java.util.ArrayList;
++import java.util.List;
++
++public class PaperLibraryStore implements LibraryStore {
++
++    private final List<Path> paths = new ArrayList<>();
++
++    @Override
++    public void addLibrary(@NotNull Path library) {
++        this.paths.add(library);
++    }
++
++    public List<Path> getPaths() {
++        return this.paths;
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/plugin/manager/MultiRuntimePluginProviderStorage.java b/src/main/java/io/papermc/paper/plugin/manager/MultiRuntimePluginProviderStorage.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ee00918a235f75b94db787933c3ab1a9abac4948
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/plugin/manager/MultiRuntimePluginProviderStorage.java
+@@ -0,0 +1,48 @@
++package io.papermc.paper.plugin.manager;
++
++import com.mojang.logging.LogUtils;
++import io.papermc.paper.plugin.entrypoint.Entrypoint;
++import io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler;
++import io.papermc.paper.plugin.provider.PluginProvider;
++import io.papermc.paper.plugin.provider.type.paper.PaperPluginParent;
++import io.papermc.paper.plugin.storage.ServerPluginProviderStorage;
++import org.bukkit.plugin.java.JavaPlugin;
++
++import java.util.ArrayList;
++import java.util.List;
++
++public class MultiRuntimePluginProviderStorage extends ServerPluginProviderStorage {
++
++    private static final org.slf4j.Logger LOGGER = LogUtils.getLogger();
++    private final List<JavaPlugin> provided = new ArrayList<>();
++
++    @Override
++    public void register(PluginProvider<JavaPlugin> provider) {
++        if (provider instanceof PaperPluginParent.PaperServerPluginProvider) {
++            LOGGER.warn("Skipping loading of paper plugin requested from SimplePluginManager.");
++            return;
++        }
++        super.register(provider);
++        /*
++        Register the provider into the server entrypoint, this allows it to show in /plugins correctly. Generally it might be better in the future to make a separate storage,
++         as putting it into the entrypoint handlers doesn't make much sense.
++         */
```

Comment:
This looks pretty scuffed. I'd either follow something like Google's style guide or just use `//`